appservice/src/intTest/groovy/nu/yona/server/ActivityTest.groovy

```diff
@@ -12,168 +12,257 @@ import java.time.ZoneId
 import java.time.ZonedDateTime
 
 import nu.yona.server.test.TimeZoneGoal
+import nu.yona.server.test.User
 
 class ActivityTest extends AbstractAppServiceIntegrationTest
 {
+	def bob, richard
+	def bobBudgetGoalGamblingUrl, bobBudgetGoalGamblingUrlForRichard, bobBudgetGoalNewsUrl, bobBudgetGoalNewsUrlForRichard, bobTimeZoneGoalUrl, bobTimeZoneGoalUrlForRichard
+	def bobDailyActivityReportsUrlForRichard, bobWeeklyActivityReportsUrlForRichard
+	def dateTime
+
+	void setupTestScenario()
+	{
+		def richardAndBob = addRichardAndBobAsBuddies()
+		richard = richardAndBob.richard
+		bob = richardAndBob.bob
+		appService.addGoal(appService.&assertResponseStatusCreated, bob, TimeZoneGoal.createInstance(SOCIAL_ACT_CAT_URL, ["11:00-12:00"].toArray()), "Going to restrict my social time!")
+		def goals = appService.getGoals(bob)
+		bobBudgetGoalGamblingUrl = findGoal(goals, GAMBLING_ACT_CAT_URL)._links.self.href
+		bobBudgetGoalNewsUrl = findGoal(goals, NEWS_ACT_CAT_URL)._links.self.href
+		bobTimeZoneGoalUrl = findGoal(goals, SOCIAL_ACT_CAT_URL)._links.self.href
+
+		def richardWithBuddy = appService.getUser(appService.&assertUserGetResponseDetailsWithPrivateData, richard.url, true, richard.password)
+		assert richardWithBuddy.buddies != null
+		assert richardWithBuddy.buddies.size() == 1
+		def bobGoalsForRichard = richardWithBuddy.buddies[0].goals
+		bobBudgetGoalGamblingUrlForRichard = bobGoalsForRichard.find{it.activityCategoryUrl == GAMBLING_ACT_CAT_URL}.url
+		bobBudgetGoalNewsUrlForRichard = bobGoalsForRichard.find{it.activityCategoryUrl == NEWS_ACT_CAT_URL}.url
+		bobTimeZoneGoalUrlForRichard = bobGoalsForRichard.find{it.activityCategoryUrl == SOCIAL_ACT_CAT_URL}.url
+		bobDailyActivityReportsUrlForRichard = richardWithBuddy.buddies[0].dailyActivityReportsUrl
+		assert bobDailyActivityReportsUrlForRichard
+		bobWeeklyActivityReportsUrlForRichard = richardWithBuddy.buddies[0].weeklyActivityReportsUrl
+		assert bobWeeklyActivityReportsUrlForRichard
+
+		addSomeTestActivity(bob)
```

Comment:
In all of our tests, Richard Quin (RQ) is making the ReQuests and and Bob Dunn (BD) is the watching BuDdy. I know it's quite a bit of work, so I hesitate to ask, but still I do: can you swap the roles of Bob and Richard here and add the activity for Richard?